### ISSUE TITLE
Easy addition of parameters for checkpointing to trainer.fit

### ISSUE THREAD

**Issue by alehuerta1** [scverse/scvi-tools#3500]
Currently, there is no straightforward way to pass a custom Lightning checkpoint containing optimizer and learning rate scheduler states to trainer.fit.

I propose updating the training runner’s  __call__ method to accept a ckpt_path argument, which can then be passed to trainer.fit for resuming training from a specific checkpoint:

    def __call__(self, ckpt_path = None):
        """Run training."""
        if hasattr(self.data_splitter, "n_train"):
            self.training_plan.n_obs_training = self.data_splitter.n_train
        if hasattr(self.data_splitter, "n_val"):
            self.training_plan.n_obs_validation = self.data_splitter.n_val
        try:
            if ckpt_path is not None:
                self.trainer.fit(self.training_plan, self.data_splitter, ckpt_path=ckpt_path)
            else:
                self.trainer.fit(self.training_plan, self.data_splitter)

**Comment by marianogabitto** [comment]
Ori, Can, 
        can we add this to the codebase? This will really help us and it is exposing a minimal number of parameters. 

Best,
Mariano

**Comment by ori-kron-wis** [comment]
@marianogabitto , @alehuerta1 
Technically, in scvi-tools you can use a callback that is called SaveCheckpoint, which is a pytorch' ModelCheckpoint wrapper to resume your work from a specific checkpoint (it can also check for runtime model crash)
e.g:
```
from scvi.model import SCVI
from scvi.train import SaveCheckpoint

adata = scvi.data.synthetic_iid()
scvi.model.SCVI.setup_anndata(
        adata,
        labels_key="labels",
        batch_key="batch",
)

model = SCVI(adata)
cb = SaveCheckpoint(
    dirpath="scvi_ckpts/",
    monitor="validation_loss",  # pick a metric your model logs
    load_best_on_end=True,      # optional: load best state into model at end
)
model.train(max_epochs=200, callbacks=[cb],enable_checkpointing=True,)

best_path = cb.best_model_path  # path for the best saved state
# later / elsewhere:
reloaded = SCVI.load(best_path, adata=adata)  # ready to use or continue training
```

However, I also added the option for the pure lightning form as you requested, and you can use that as well.
e.g this will work now, following the same code as above, and using pytorch' ModelCheckpoint:
`model.train(max_epochs=1, check_val_every_n_epoch=1, ckpt_path="best")
`

You can add a unit test to the PR to match your scenario as well.

**Comment by marianogabitto** [comment]
Ori, 
       I don't think that the way you are using checkpoint in scvi, neither save nor load the parameters and the state of the optimizer. We developed a way, convoluted so it does not merit a post, to save the optimizer state and load it if we have this option.

     We will add a unit test !

Thanks,
M

**Comment by ori-kron-wis** [comment]
Indeed, we do not save or load the optimizer state in this mechanism.
Please contribute to this PR.

**Comment by ori-kron-wis** [comment]
@alehuerta1 Once we will merge the PR this issue will automatically close. 
I left you a small remark there.
